src/dolphin/unwrap/_whirlwind.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+import logging
+from contextlib import ExitStack
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+
+from dolphin._types import Filename
+from dolphin.io._core import DEFAULT_TIFF_OPTIONS_RIO
+from dolphin.utils import full_suffix
+
+from ._constants import CONNCOMP_SUFFIX, DEFAULT_CCL_NODATA, DEFAULT_UNW_NODATA
+from ._utils import _zero_from_mask
+
+__all__ = [
+    "unwrap_whirlwind",
+]
+
+
+logger = logging.getLogger(__name__)
+
+
+def unwrap_whirlwind(
+    ifg_filename: Filename,
+    corr_filename: Filename,
+    unw_filename: Filename,
+    nlooks: float,
+    mask_file: Optional[Filename] = None,
+    zero_where_masked: bool = False,
+    unw_nodata: Optional[float] = DEFAULT_UNW_NODATA,
+    ccl_nodata: Optional[int] = DEFAULT_CCL_NODATA,
+    scratchdir: Optional[Filename] = None,
+) -> tuple[Path, Path]:
+    """Unwrap an interferogram using `whirlwind`.
+
+    Parameters
+    ----------
+    ifg_filename : Filename
+        Path to input interferogram.
+    corr_filename : Filename
+        Path to input correlation file.
+    unw_filename : Filename
+        Path to output unwrapped phase file.
+    nlooks : float
+        Effective number of looks used to form the input correlation data.
+    mask_file : Filename, optional
+        Path to binary byte mask file, by default None.
+        Assumes that 1s are valid pixels and 0s are invalid.
+    zero_where_masked : bool, optional
+        Set wrapped phase/correlation to 0 where mask is 0 before unwrapping.
+        If not mask is provided, this is ignored.
+        By default True.
+    unw_nodata : float, optional
+        If providing `unwrap_callback`, provide the nodata value for your
+        unwrapping function.
+    ccl_nodata : float, optional
+        Nodata value for the connected component labels.
+    scratchdir : Filename, optional
+        If provided, uses a scratch directory to save the intermediate files
+        during unwrapping.
+
+    Returns
+    -------
+    unw_path : Path
+        Path to output unwrapped phase file.
+    conncomp_path : Path
+        Path to output connected component label file.
+
+    """
+    import snaphu
+    import whirlwind as ww
+
+    # Create a context manager that combines other context managers -- one for each
+    # input raster file. Upon exiting the context block, each context manager in the
+    # stack will be closed in LIFO order.
+    with ExitStack() as stack:
```

Comment:
This is the better version that you mentioned that `snaphu-py` should also use [instead of this?](https://github.com/isce-framework/dolphin/blob/main/src/dolphin/unwrap/_snaphu_py.py#L105-L107)

Reply:
Yeah, that's right